cpp/include/raft/neighbors/detail/cagra/cagra_build.cuh

```diff
@@ -344,6 +345,16 @@ index<T, IdxT> build(
   RAFT_LOG_INFO("Graph optimized, creating index");
   // Construct an index from dataset and optimized knn graph.
   if (construct_index_with_dataset) {
+    if (params.compression.has_value()) {
+      index<T, IdxT> idx(res, params.metric);
+      idx.update_graph(res, raft::make_const_mdspan(cagra_graph.view()));
+      idx.update_dataset(
+        res,
+        // TODO: ATM, only float math type is supported in kmeans training.
```

Comment:
Note that kmeans can take int8 input, but under the hood it maps to float. This will improve in the future.